src/main/java/org/commcare/core/network/ModernHttpRequester.java

```diff
@@ -41,149 +40,93 @@
         }
     }
 
-    private URL buildUrl() throws IOException {
-        if (isPostRequest) {
-            return url;
-        } else {
-            return buildUrlWithParams();
-        }
-    }
-
-    protected HttpURLConnection setupConnection(URL builtUrl) throws IOException {
-        HttpURLConnection httpConnection = (HttpURLConnection)builtUrl.openConnection();
-        setupConnectionInner(httpConnection);
-        if (isPostRequest) {
-            httpConnection.setRequestMethod("POST");
-            httpConnection.setDoOutput(true);
-            buildPostPayload(httpConnection);
-        } else {
-            httpConnection.setRequestMethod("GET");
+    /**
+     * Executes the HTTP Request. Can be called directly to bypass response processor.
+     *
+     * @return Response from the HTTP call
+     * @throws IOException
+     */
+    public Response<ResponseBody> makeRequest() throws IOException {
+        switch (method) {
+            case POST:
+                currentCall = commCareNetworkService.makePostRequest(url, params, addContentHeaders(requestBody), requestBody);
+                break;
+            case MULTIPART_POST:
+                currentCall = commCareNetworkService.makeMultipartPostRequest(url, params, headers, parts);
+                break;
+            case GET:
+                currentCall = commCareNetworkService.makeGetRequest(url, params, headers);
+                break;
+            default:
+                throw new IllegalArgumentException("Invalid HTTPMethod " + method.toString());
         }
-
-        return httpConnection;
-    }
-
-    private static void setupConnectionInner(HttpURLConnection httpConnection) {
-        httpConnection.setConnectTimeout(CONNECTION_TIMEOUT);
-        httpConnection.setReadTimeout(CONNECTION_SO_TIMEOUT);
-        httpConnection.setDoInput(true);
-        httpConnection.setInstanceFollowRedirects(true);
+        return currentCall.execute();
     }
 
-    private void buildPostPayload(HttpURLConnection httpConnection) throws IOException {
-        String paramsString = buildUrlWithParams().getQuery();
-        int bodySize = paramsString.length();
-        httpConnection.setRequestProperty("Content-Type", "application/x-www-form-urlencoded");
-        httpConnection.setRequestProperty("Content-Length", bodySize + "");
-        // write to connection
-        OutputStream os = httpConnection.getOutputStream();
-        BufferedWriter writer = new BufferedWriter(
-                new OutputStreamWriter(os, "UTF-8"));
-        writer.write(paramsString);
-        writer.flush();
-        writer.close();
-        os.close();
-    }
 
-    protected URL buildUrlWithParams() throws MalformedURLException {
-        UriBuilder b = UriBuilder.fromUri(url.toString());
-        for (Map.Entry<String, String> param : params.entrySet()) {
-            b.queryParam(param.getKey(), param.getValue());
+    private Map<String, String> addContentHeaders(RequestBody postBody) {
```

Comment:
This method name seems like it's confusingly generic. It's "addContentHeadersForXFormPost", right? Not a generic thing

Reply:
Yeah. Also the only Post request we make is from `PostRequestActivity` and I was not sure of use case for that. Since it's causing confusion I think it makes sense to move this method to `PostRequestActivity` itself and rename it to `addContentHeadersForXFormPost`

Reply:
I think either would be good, would just want it to be clear from some form of context that the content header added here is going to set a mime type to an XForm for posting.